Gitweb:     http://git.kernel.org/linus/8629ea2eaba8ca0de2e38ce1b4a825e16255976e
Commit:     8629ea2eaba8ca0de2e38ce1b4a825e16255976e
Parent:     8747d793fc5c4d3e4decd41d55f6dc24498dd5f5
Author:     Feng Tang <feng.tang@intel.com>
AuthorDate: Thu Sep 3 16:32:53 2009 +0800
Committer:  Thomas Gleixner <tglx@linutronix.de>
CommitDate: Fri Nov 20 11:25:48 2009 +0100

    hrtimer: Fix /proc/timer_list regression
    
    commit 507e1231 (timer stats: Optimize by adding quick check to avoid
    function calls) introduced a regression in /proc/timer_list.
    
    /proc/timer_list shows now
     #0: <c27d46b0>, tick_sched_timer, S:01, <(null)>, /-1
    instead of
     #0: <c27d46b0>, tick_sched_timer, S:01, hrtimer_start, swapper/0
    
    Revert the hrtimer quick check for now. The optimization needs more
    thought, but this is neither 2.6.32-rc7 nor stable material.
    
    [ tglx: - Removed unrelated changes from the original patch
      	- Prevent unneccesary call to timer_stats_update_stats
    	- massaged the changelog ]
    
    Signed-off-by: Feng Tang <feng.tang@intel.com>
    LKML-Reference: <alpine.LFD.2.00.0911181933540.24119@localhost.localdomain>
    Cc: Heiko Carstens <heiko.carstens@de.ibm.com>
    Cc: stable@kernel.org
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 include/linux/hrtimer.h |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

--- a/include/linux/hrtimer.h
+++ b/include/linux/hrtimer.h
@@ -446,7 +446,7 @@ extern void timer_stats_update_stats(voi
 
 static inline void timer_stats_account_hrtimer(struct hrtimer *timer)
 {
-	if (likely(!timer->start_site))
+	if (likely(!timer_stats_active))
 		return;
 	timer_stats_update_stats(timer, timer->start_pid, timer->start_site,
 				 timer->function, timer->start_comm, 0);
@@ -457,8 +457,6 @@ extern void __timer_stats_hrtimer_set_st
 
 static inline void timer_stats_hrtimer_set_start_info(struct hrtimer *timer)
 {
-	if (likely(!timer_stats_active))
-		return;
 	__timer_stats_hrtimer_set_start_info(timer, __builtin_return_address(0));
 }
 
